### PR TITLE
feat: init ldmk-transport pkg

### PR DESCRIPTION
**Pull request by fAnselmi-Ledger**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - ...

### 📝 Description

Created LDMK transport package

### ❓ Context

- **JIRA or GitHub link**: [DSDK-506](https://ledgerhq.atlassian.net/browse/DSDK-506)

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[DSDK-506]: https://ledgerhq.atlassian.net/browse/DSDK-506?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/ldmk-transport/package.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
+{
+  "name": "@ledgerhq/ldmk-transport",
```
Comment (jdabbech-ledger) [inline libs/ldmk-transport/package.json:2]: shouldn't it be named `@ledgerhq/live-dmdk` ?

File: libs/ldmk-transport/src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  DeviceSdkBuilder,
+  ConsoleLogger,
+  type DeviceSdk,
+  type DeviceSessionState,
+  DeviceStatus,
+  LogLevel,
+} from "@ledgerhq/device-management-kit";
+import { BehaviorSubject, firstValueFrom } from "rxjs";
+
+const deviceSdk = new DeviceSdkBuilder().addLogger(new ConsoleLogger(LogLevel.Debug)).build();
```
Comment (jdabbech-ledger) [inline libs/ldmk-transport/src/index.tsx:13]: The dmdk transport integration dmdk is available on develop.
You should use the latest version and `builder.addCustomTransport(DMDKTransport)` here with a transport defined as `class DMDKTransport extends Transport`

**Review by jdabbech-ledger — CHANGES_REQUESTED** [review]

File: libs/ldmk-transport/src/index.tsx (hunk 2 above)
Comment (ofreyssinet-ledger) [reply libs/ldmk-transport/src/index.tsx:13]: @jdabbech-ledger it's a LL `Transport` here not a LDMK transport

Comment (ofreyssinet-ledger) [reply libs/ldmk-transport/src/index.tsx:13]: It's a transport like other ledger live transports, but it's a wrapper around the LDMK.

Screenshot 2024-09-26 at 12 02 07

File: libs/ldmk-transport/package.json (hunk 1 above)
Comment (ofreyssinet-ledger) [reply libs/ldmk-transport/package.json:2]: indeed the package should be called `@ledgerhq/live-dmk`

File: libs/ldmk-transport/src/index.tsx (hunk 2 above)
Comment (jdabbech-ledger) [reply libs/ldmk-transport/src/index.tsx:13]: Oh ok, jira leaks information on this part. 
Will you add here a node hid transport afterwards?

File: libs/ldmk-transport/src/index.tsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    const activeSessionId = activeDeviceSessionSubject.value?.sessionId;
+
+    if (activeSessionId) {
+      console.log("[SDKTransport][open] checking existing session", activeSessionId);
+      const deviceSessionState: DeviceSessionState | null = await firstValueFrom(
+        deviceSdk.getDeviceSessionState({ sessionId: activeSessionId }),
+      ).catch(e => console.error("[SDKTransport][open] error getting device session state", e));
+
+      if (deviceSessionState?.deviceStatus !== DeviceStatus.NOT_CONNECTED) {
+        console.log(
```
Comment (jdabbech-ledger) [inline libs/ldmk-transport/src/index.tsx:112]: [could] use LocalTracer logs

File: libs/live-dmk/package.json [hunk 4]
```diff
@@ -0,0 +1,68 @@
+{
+  "name": "@ledgerhq/live-dmk",
+  "version": "0.1.0",
+  "description": "Ledger Device Managment Kit transport module",
```
Comment (ofreyssinet-ledger) [inline libs/live-dmk/package.json:4]: I didn't see this but let's change it later, the description is not really accurate.
Also I think we can put `"private": true`

File: libs/live-dmk/src/index.tsx [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+      });
+
+      if (deviceSessionState?.deviceStatus !== DeviceStatus.NOT_CONNECTED) {
+        this.tracer.trace("[open] reusing existing session and instantiating a new SdkTransport");
+        return activeDeviceSessionSubject.value.transport;
+      }
+    }
+
+    this.tracer.trace("[open] No active session found, starting discovery");
+    const discoveredDevice: { id: string } = await firstValueFrom(deviceSdk.startDiscovering());
```
Comment (ofreyssinet-ledger) [inline libs/live-dmk/src/index.tsx:120]: const [discoveredDevice] = await firstValueFrom(deviceSdk.listenToKnownDevices());

File: libs/live-dmk/package.json [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    "build": "tsc && tsc -m ES6 --outDir lib-es",
+    "prewatch": "pnpm build",
+    "watch": "tsc --watch",
+    "lint": "eslint ./src --no-error-on-unmatched-pattern --ext .ts,.tsx --cache",
+    "lint:fix": "pnpm lint --fix",
+    "typecheck": "tsc --noEmit",
+    "unimported": "unimported",
+    "test": "jest"
+  },
+  "workspaces": [
```
Comment (valpinkman) [inline libs/live-dmk/package.json:32]: [ASK] why do we need this ?

**Review by valpinkman — APPROVED** [review]

File: libs/live-dmk/package.json [hunk 7]
```diff
@@ -0,0 +1,65 @@
+{
+  "name": "@ledgerhq/live-dmk",
+  "version": "0.1.0",
+  "description": "Ledger Device Managment Kit transport module",
```
Comment (ofreyssinet-ledger) [inline libs/live-dmk/package.json:4]: [SHOULD] typo: Management

Comment (valpinkman) [reply libs/live-dmk/package.json]: fixed

**Review by ofreyssinet-ledger — DISMISSED** [review]

**Review by valpinkman — APPROVED** [review]

**Pull request merged**